cvat/apps/engine/media_extractors.py

```diff
@@ -179,25 +194,48 @@ def _make_name():
 
 class ZipReader(ImageListReader):
     def __init__(self, source_path, step=1, start=0, stop=None):
-        self._zip_source = zipfile.ZipFile(source_path[0], mode='r')
+        self._zip_source = zipfile.ZipFile(source_path[0], mode='a')
         self.extract_dir = source_path[1] if len(source_path) > 1 else None
         file_list = [f for f in self._zip_source.namelist() if get_mime(f) == 'image']
         super().__init__(file_list, step, start, stop)
 
     def __del__(self):
         self._zip_source.close()
 
-    def get_preview(self):
+    def get_preview(self, dimension="2d"):
```

Comment:
From my point of view it would be better to inherit a new class to support 3d data. In this case it's not needed to add dummy arguments (I mean `dimension`) to the other readers that will never support them.
You can determinate the proper reader here https://github.com/openvinotoolkit/cvat/blob/0fd3742c5a50d55e09258a5dc241ce91d72c3bc4/cvat/apps/engine/task.py#L109 by using your `ValidateDimention` class.

Reply:
Currently there is an existing mime-type for pcd i.e 'image/x-photo-cd pcd' in mime-types, hence we are distinguishing the 3D and 2D based on headers in the PCD file, if the existing mime-type can be deleted then we can implement in the similar manner mentioned.

Reply:
Let's get rid of the old one

Reply:
Made changes accordingly